src/soundio/sounddevicenetwork.cpp

```diff
@@ -43,31 +52,45 @@ SoundDeviceError SoundDeviceNetwork::open(bool isClkRefDevice, int syncBuffers)
 
     // Get latency in milleseconds
     qDebug() << "framesPerBuffer:" << m_framesPerBuffer;
-    double bufferMSec = m_framesPerBuffer / m_dSampleRate * 1000;
-    qDebug() << "Requested sample rate: " << m_dSampleRate << "Hz, latency:"
-             << bufferMSec << "ms";
 
-    // Create the callback function pointer.
-    if (isClkRefDevice) {
-        // Network device as clock Reference is not yet supported
-        DEBUG_ASSERT(false);
-    } else {
-        // Feet the network device buffer directly from the
-        // clock reference device callback
-        // This is what should work best.
+    m_audioBufferTime = mixxx::Duration::fromSeconds(
+            m_framesPerBuffer / m_dSampleRate);
+    qDebug() << "Requested sample rate: " << m_dSampleRate << "Hz, latency:"
+             << m_audioBufferTime;
 
-        if (m_iNumOutputChannels) {
-            m_outputFifo = new FIFO<CSAMPLE>(
-                    m_iNumOutputChannels * m_framesPerBuffer * 2);
-        }
-        if (m_iNumInputChannels) {
-            m_inputFifo = new FIFO<CSAMPLE>(
-                    m_iNumInputChannels * m_framesPerBuffer * 2);
-        }
+    // Feet the network device buffer directly from the
+    // clock reference device callback
+    // This is what should work best.
+    if (m_iNumOutputChannels) {
+        m_outputFifo = std::make_unique<FIFO<CSAMPLE> >(
+                m_iNumOutputChannels * m_framesPerBuffer * 2);
+    }
+    if (m_iNumInputChannels) {
+        m_inputFifo = std::make_unique<FIFO<CSAMPLE> >(
+                m_iNumInputChannels * m_framesPerBuffer * 2);
     }
 
     m_pNetworkStream->startStream(m_dSampleRate);
 
+    // Create the callback Thread if requested
+    if (isClkRefDevice) {
+        // Update the samplerate and latency ControlObjects, which allow the
+        // waveform view to properly correct for the latency.
+        ControlObject::set(ConfigKey("[Master]", "latency"),
```

Comment:
Using ControlObjects for this feels like an ugly hack. Should there be a TODO to pass this information directly to the waveform renderers without requiring a ControlObject?

Reply:
The code was adopted from here:
https://github.com/mixxxdj/mixxx/blob/2b26c4697cbaae7960975fab06274912bd16cda4/src/soundio/sounddeviceportaudio.cpp#L369
And I do not think this is a bad solution, because this way, the controllers can use this info as well.
(Like LPD8RK)

Reply:
Controller scripts should not need this information. The Akai LPD 8 script seems to use this to hack beatjumping in JS, but we have that in C++ now.

Reply:
I can update the LPD 8 mapping to fix that. Thanks for the reminder about that controller.